client/src/components/VideoPlayer.jsx

```diff
@@ -1,78 +1,87 @@
 import React from 'react';
-
+import PlayPauseButton from './PlayPauseButton.jsx'
+import MuteButton from './MuteButton.jsx'
 // there are two file types when using videos on browser. MP4 and WEBM is widely used across most if not all browsers. So its good to have both file types for your video tag in the event the browser can't load one. If both cannot load you can include a text to display (similar to the alt attribute for img tags).
 
 // In creating the video tag, in order to get it to autoplay I couldn't get it to work with just autoplay. It worked when I changed it to autoplay='autoplay'. But one thing to note is that React is warning me about autoplay not being camel-cased and recognized (iif it is camel-cased). Also Chrome requires you to mute the video in order to autoplay (but I think its muted by default now)
 class VideoPlayer extends React.Component {
     constructor(props) {
         super(props)
         this.state = {
-            paused: false
+            paused: false,
+            volume: 0,
+            tempVolume: .1
         }
     }
 
-    play(e) {
+    togglePlayPause(e) {
         const video = document.getElementById("video-player");
-
         if (this.state.paused == true) {
-          video.play();
-          this.setState({
-            paused: false
-          });
+            this.setState({
+                paused: false
+            }, () => {
+                video.play();
+            });
         } else {
-          video.pause();
-          this.setState({
-            paused: true
-          });
+            this.setState({
+                paused: true
+            }, () => {
+                video.pause();
+            });
+        }
+    }
+
+    handleVolume(e) {
+        const video = document.getElementById("video-player");
+        let tempValue = e.target.value
+        if (e.target.value < 0.01) {
+            tempValue = .1
         }
-      }
+        this.setState({
+            volume: e.target.value,
+            tempVolume: tempValue
+        }, () => {
+            video.volume = this.state.volume
+        })
+
+        if (e.target.value > 0){
+            video.muted = false
+        } else if (e.target.value < 0.01) {
+            video.muted = true
+        }
+    }
+
+    toggleMute(e) {
+        const video = document.getElementById("video-player");
+        if (this.state.volume > 0) {
+            this.setState({
+                volume: 0
+            });
+            video.muted = true
+        } else if (this.state.volume < 0.01) {
+            this.setState({
+                volume: this.state.tempVolume
+            }, () => {
+                video.muted = false
+                video.volume = this.state.volume
```

Comment:
Hmmm... I don't follow this logic. I'll have to try out the mute and see how it works. I would think this would be more like:
video.muted = !video.muted

Reply:
I wanted to mimic the twitch UI. So what happens is when the user makes the volume go down to 0 it will set a default value of .1 (for whatever reason doing this.state.volume === 0 was not working, hence the <.01). That was when the user clicks the button it doesnt stay at 0.

Reply:
Now that I look at it I dont need to do anything  to tempValue other than have it set to .1 in the state. This is in referrence to the volume function.